### PR TITLE
Feature/emlys/doc 654

### PR DESCRIPTION
**Pull request by emlys**
dockstore/dockstore#2340
- minor changes removing redundant parentheses etc.
- updated quay.io locations to reflect expected 1.0.0 release
- replaced `return` with `sys.exit` in check_md5sum.py so that exit codes are visible outside the file

### CODE REVIEW

**Review by denis-yuen — COMMENTED** [review]
Would be nice to have some rudimentary CI testing. 
Something like of https://github.com/briandoconnor/dockstore-tool-md5sum/blob/master/.travis.yml (without the Slack bits) but with the currently recommended version of cwltool

**Comment by emlys** [comment]
To demo:
the docker images in checker/md5sum-checker.cwl and checker/md5sum-checker.wdl do not yet exist as they refer to an anticipated first release. to test, replace `quay.io/dockstore-testing/md5sum-checker:1.0.0` with `quay.io/dockstore-testing/md5sum-checker:feature_emlys_DOC-654` in both files.

run `java -jar cromwell-42.jar run checker-workflow-wrapping-workflow.wdl -i md5sum-wdl.json` and `cwltool checker-workflow-wrapping-tool.cwl checker-input-cwl.json`. both should succeed.

make an edit to the sample input file, md5sum.input. 

rerun the commands and both should fail.

**Comment by emlys** [comment]
> Would be nice to have some rudimentary CI testing.
> Something like of https://github.com/briandoconnor/dockstore-tool-md5sum/blob/master/.travis.yml (without the Slack bits) but with the currently recommended version of cwltool

Added a test file and just got it to work in Travis

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+python:
+  - "2.7"
+
+before_install:
+  # Download Cromwell
+  - wget -O $TRAVIS_BUILD_DIR/cromwell-42.jar https://github.com/broadinstitute/cromwell/releases/download/42/cromwell-42.jar;
+
+install:
+  - pip install cwl-runner cwltool==1.0.20160325210917 schema-salad==1.7.20160316203940 avro==1.9.0
```
Comment (denis-yuen) [inline .travis.yml]: Let's match the version of cwltool at https://dockstore.org/api/metadata/runner_dependencies?client_version=1.6.0&python_version=2

~Double check that cromwell matches the version that the CLI installs as well~
Actually, looks like we're using an older version for some reason. Nevermind the above

File: test.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import subprocess
+import unittest
+
+
+class TestMd5sumChecker(unittest.TestCase):
```
Comment (denis-yuen) [inline test.py:5]: :+1:

File: .travis.yml (hunk 1 above)
Comment (emlys) [reply .travis.yml]: Okay cool

Comment (emlys) [reply .travis.yml]: I just updated it to python3.6 and newest versions of each requirement

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+python:
+  - "3.6"
+
+before_install:
+  # Download Cromwell
+  - wget -O $TRAVIS_BUILD_DIR/cromwell-42.jar https://github.com/broadinstitute/cromwell/releases/download/42/cromwell-42.jar;
+
+install:
+  - pip install cwl-runner cwltool==1.0.20190621234233 schema-salad==4.5.20190621200723 avro==1.9.0
```
Comment (garyluu) [inline .travis.yml]: In my opinion, it's probably better to follow the tutorial

```
curl -o requirements.txt "https://dockstore.org/api/metadata/runner_dependencies?client_version=1.6.0&python_version=3"
pip install -r requirements.txt
```
except instead of using the live dockstore.org, use https://raw.githubusercontent.com/dockstore/dockstore/1.6.0/dockstore-webservice/src/main/resources/requirements/1.6.0/requirements3.txt or something

File: checker/md5sum-checker.wdl [hunk 4]
```diff
@@ -7,6 +7,6 @@ task checkerTask {
   }
 
   runtime {
-    docker: "quay.io/agduncan94/checker-md5sum"
+    docker: "quay.io/dockstore-testing/md5sum-checker:feature_emlys_DOC-654"
```
Comment (garyluu) [inline checker/md5sum-checker.wdl:10]: what changed in this image? can you include the new Dockerfile

Comment (emlys) [reply checker/md5sum-checker.wdl:10]: the Dockerfile hasn't changed, just the check_md5sum.py file that's gotten from this new image

File: .travis.yml (hunk 3 above)
Comment (emlys) [reply .travis.yml]: okay. it should be python 2?

Comment (garyluu) [reply .travis.yml]: 3 is fine, i'll edit my example

**Comment by garyluu** [comment]
Others should probably chime in.  But I'd prefer the tests to fail if I modify checker/bin/check_md5sum to something completely nonsensical.  This may involve something fairly hacky like:
```
docker build -t potato -f checker/Dockerfile checker
sed 's/dockerPull:.*/dockerPull: potato:latest/g' checker/md5sum-checker.cwl
sed 's/docker:.*/docker: "potato:latest"/g' checker/md5sum-checker.wdl 
```
before running the test

File: .travis.yml (hunk 3 above)
Comment (emlys) [reply .travis.yml]: Thanks, just changed this

**Review by denis-yuen — APPROVED** [review]

File: checker/md5sum-checker.cwl [hunk 5]
```diff
@@ -7,7 +7,7 @@ cwlVersion: v1.0
 
 requirements:
 - class: DockerRequirement
-  dockerPull: quay.io/agduncan94/checker-md5sum
+  dockerPull: quay.io/dockstore-testing/md5sum-checker:feature_emlys_DOC-654
```
Comment (coverbeck) [inline checker/md5sum-checker.cwl:10]: This is a chicken and egg situation, but I think you want the version to be `1.0.0`, which will only exist after this PR is merged and you have created a 1.0.0 GitHub release. Hmm, but then your tests will fail.

Maybe Gary's self-described "hacky" idea in another comment is the way to go.

File: checker/md5sum-checker.wdl (hunk 4 above)
Comment (coverbeck) [inline checker/md5sum-checker.wdl:10]: Will need to be version 1.0.0.

File: checker/bin/check_md5sum [hunk 6]
```diff
@@ -6,39 +6,41 @@ import json
...
-    results_file.write('{ "overall" : false, "steps" : { "md5sum_check" : false } }')
-  results_file.close()
+    log_file = open("log.txt", "w")
+    log_file.write("")
+    log_file.close()
 
-  log_file  = open("log.txt", "w")
-  log_file.write("")
-  log_file.close()
+    sys.exit(result_code)
```
Comment (coverbeck) [inline checker/bin/check_md5sum:29]: As we discussed, I think this is the only line that matters for the fix in this file. In a perfect world, the PR would probably only have this line, and the formatting would done in a separate PR... But not worth changing.

**Comment by denis-yuen** [comment]
I'm ok with more than one PR in order to reach a particular destination. 
This clearly looks like an improvement, so I would recommend reaching a good point, merging, tagging a release, then fixing the version numbers, giving Gary's idea a shot, etc.

**Review by coverbeck — APPROVED** [review]
